postprocessor/src/main/java/com/azure/autorest/postprocessor/util/PartialUpdateHandler.java

```diff
@@ -160,11 +164,164 @@ private static String handlePartialUpdateForClassOrInterfaceFile(String generate
+
+        NodeList<ModuleDirective> moduleDirectives = new NodeList<>();
+        moduleDirectives.addAll(requiresDirectiveNodeList);
+        moduleDirectives.addAll(exportsDirectiveNodeList);
+        moduleDirectives.addAll(opensDirectiveNodeList);
+        moduleDirectives.addAll(usesDirectiveNodeList);
+        moduleDirectives.addAll(providesDirectiveNodeList);
+
+        ModuleDeclaration moduleDeclaration = compilationUnitForGeneratedFile.getModule().get();
+        moduleDeclaration.setDirectives(moduleDirectives);
+
+        compilationUnitForGeneratedFile.setModule(moduleDeclaration);
+
+        return compilationUnitForGeneratedFile.toString();
+    }
+
+
+    /**
+     *
+     * Merge two directive list. The logic is as below:
+     * 1. Add all the directives in list1 to the returned list.
+     * 2. For each directive in list2, check if it is in list1, if it is in list1, then we don't need to add it to returned list, otherwise, we need to add it to the returned list
+     * @param list1 first directive list
+     * @param list2 second directive list
+     * @return the merged directive list
+     */
+    private static NodeList<ModuleDirective> mergeDirectiveNodeList(NodeList<ModuleDirective> list1, NodeList<ModuleDirective> list2) {
+        NodeList<ModuleDirective> res = new NodeList<>();
+        res.addAll(list1);
+        for (ModuleDirective directive2 : list2) {
+            boolean isInList1 = false;
+            for (ModuleDirective directive1 : list1) {
+                if (directive1.getTokenRange().isPresent() && directive2.getTokenRange().isPresent()) {
+                    // 1. build two token lists from the two directives, only put in non-empty tokens
+                    // 2. compare the tokens one by one in the token lists
+                    // 3. if the two token lists are the same, then we consider directive2 is in list1, otherwise, we consider directive2 is not in list1
+                    List<JavaToken> tokenList1 = new ArrayList<>();
+                    List<JavaToken> tokenList2 = new ArrayList<>();
+                    for (JavaToken token1 : directive1.getTokenRange().get()) {
+                        if (!token1.asString().trim().isEmpty()) {
+                            tokenList1.add(token1);
+                        }
+                    }
+                    for (JavaToken token2 : directive2.getTokenRange().get()) {
+                        if (!token2.asString().trim().isEmpty()) {
+                            tokenList2.add(token2);
+                        }
+                    }
+                    if (tokenList1.size() == tokenList2.size()) {
+                        boolean isTokenList1EqualsTokenList2 = true;
+                        for (int i = 0; i < tokenList1.size(); ++i) {
+                            if (!tokenList1.get(i).asString().trim().equals(tokenList2.get(i).asString().trim())) {
```

Comment:
Why not `.asString().trim()` then add to list, since you do it anyway in the `if`?

Then you seems only need to compare 2 list with "equals" (not its element one-by-one)?

Reply:
Adding tokens as List<String> works, I will modify the code for simplicity